maxfw/tests/test_image_utils.py

```diff
@@ -0,0 +1,400 @@
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert img_out.shape == (200, 200, 4)
+
+    # Test that the values in all 4 output channels are identical
+    assert img_out[..., 0].all() == img_out[..., 1].all() == img_out[..., 2].all() == img_out[..., 3].all()
+
+
+def test_imageprocessor_normalize():
+    """Test the Imageprocessor's normalize function."""
+
+    # Test normalize
+    transform_sequence = [ToPILImage('RGBA'), Normalize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.max(img_out) <= 1 and np.min(img_out) >= 0
+
+    # Test normalize
+    transform_sequence = [ToPILImage('RGB'), Normalize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.max(img_out) <= 1 and np.min(img_out) >= 0
+
+    # Test normalize
+    transform_sequence = [ToPILImage('L'), Normalize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.max(img_out) <= 1 and np.min(img_out) >= 0
+
+    # Test for wrong use
+    transform_sequence = [ToPILImage('L'), Normalize(), Resize(size=(200, 200))]
+    p = ImageProcessor(transform_sequence)
+    with nose.tools.assert_raises(Exception):
+        p.apply_transforms(test_input)
+
+    # Test for wrong use
+    transform_sequence = [ToPILImage('RGBA'), Normalize(), Normalize()]
+    p = ImageProcessor(transform_sequence)
+    with nose.tools.assert_raises(Exception):
+        p.apply_transforms(test_input)
+
+
+def test_imageprocessor_standardize():
+    """Test the Imageprocessor's standardize function."""
+
+    # Test standardize (RGBA)
+    transform_sequence = [ToPILImage('RGBA'), Standardize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    nose.tools.assert_almost_equal(np.std(img_out[..., :3]), 1)
+    nose.tools.assert_equal(np.std(img_out[..., 3]), 0)
+
+    # generate an image array
+    pil_img = ImageProcessor([ToPILImage('RGBA'), PILtoarray()]).apply_transforms(test_input)
+
+    # Test standardize (RGBA)
+    transform_sequence = [ToPILImage('RGBA'), Standardize(mean=np.mean(pil_img))]
+    ImageProcessor(transform_sequence).apply_transforms(test_input)
+
+    # Test standardize (RGBA)
+    transform_sequence = [ToPILImage('RGBA'), Standardize(mean=np.mean(pil_img), std=np.std(pil_img))]
+    ImageProcessor(transform_sequence).apply_transforms(test_input)
+
+    # Test standardize (RGBA)
+    transform_sequence = [ToPILImage('RGBA'), Standardize(std=np.std(pil_img))]
+    ImageProcessor(transform_sequence).apply_transforms(test_input)
+
+    # Test standardize (RGBA)
+    transform_sequence = [ToPILImage('RGBA'), Standardize(mean=127, std=5)]
+    ImageProcessor(transform_sequence).apply_transforms(test_input)
+
+    # Test standardize error (RGBA)
+    transform_sequence = [ToPILImage('RGBA'), Standardize(mean=[127, 127], std=5)]
```

Comment:
In these tests, are we testing channel input (ie. `[a, b, c, d]` in this case)? I only see single-values for `mean` and `std` and error cases.

Reply:
Added! Thank you Nick.